lib/utils.nix

```
@@ -172,6 +172,40 @@ rec {
     in
     lib.literalExpression exp;
 
+  /**
+    Convert one or several conditional warnings to a final warning list.
+
+    # Example
+
+    ```nix
+    warnings = mkWarnings [
+      {
+        condition = plugins.foo.settings.barIntegration && (!plugins.bar.enable);
+        message = "`barIntegration` is enabled but the `bar` plugin is not."
+      }
+    ];
+    ```
+
+    # Type
+    ```
+    mkWarnings :: String -> List -> List
+    ```
+  */
+  mkWarnings =
+    scope: w:
+    let
+      warnings = if !lib.isList w then [ w ] else w;
```

Can be simplified to:
```suggestion
      warnings = lib.toList w;
```